packages/utils/telemetry-utils/src/eventEmitterWithErrorHandling.ts

```diff
@@ -7,8 +7,12 @@ import { IEvent } from "@fluidframework/core-interfaces";
 
 /**
  * Event Emitter helper class
+ *
+ * @remarks
  * Any exceptions thrown by listeners will be caught and raised through "error" event.
  * Any exception thrown by "error" listeners will propagate to the caller.
+ *
+ * @public
  */
 export class EventEmitterWithErrorHandling<
```

Comment:
Not sure this even belongs here.  Should probably move to core-utils and be internal.

Reply:
+1. One small complication here is that [`SharedObjectCore` extends it](https://github.com/microsoft/FluidFramework/blob/5581bb24ea118ad8660a262b35a5337514c3ced4/packages/dds/shared-object-base/src/sharedObject.ts#L44-L46), `SharedObject` extends that, and our consumers extend `SharedObject`. I don't think they ever check for `instanceof EventEmitterWithErrorHandling` or similar so it might be ok to eventually move everything to a copy of this in core-utils.

Reply:
We've decided to aggressively annotate everything as `@internal` for now and do the exercise of promoting things to `@alpha` and `@public` as needed in the future. I did add some privateRemarks indicating this should probably move to core-utils, though.